.github/workflows/ci_vtests.yml

```diff
@@ -4,47 +4,239 @@ on:
     - name: Cancel Previous Runs
       uses: styfle/cancel-workflow-action@0.11.0
       with:
         access_token: ${{ github.token }}
     - name: Clone repository
-      uses: actions/checkout@v3  
+      uses: actions/checkout@v3
     - name: "Configure workflow"
-      env:
-        pull_request_title: ${{ github.event.pull_request.title }}
       run: |
-        bash ./build/ci/tools/make_build_number.sh
-        BUILD_NUMBER=$(cat ./build.artifacts/env/build_number.env)      
-
-        ADD_INFO="_${GITHUB_REF#refs/heads/}"
-        if [ "${{ github.event_name }}" == "pull_request" ]; then ADD_INFO="_${{ github.event.pull_request.number }}_${pull_request_title}"; fi
-        UPLOAD_ARTIFACT_NAME="$(tr '":<>|*?/\\' '_' <<<"VTests_Comparison_${BUILD_NUMBER}_${ADD_INFO}")"
+        sudo bash ./build/ci/tools/make_build_number.sh
+        BUILD_NUMBER=$(cat ./build.artifacts/env/build_number.env)
 
         echo "BUILD_NUMBER=$BUILD_NUMBER" >> $GITHUB_ENV
         echo "BUILD_NUMBER: $BUILD_NUMBER"
-        echo "UPLOAD_ARTIFACT_NAME=$UPLOAD_ARTIFACT_NAME" >> $GITHUB_ENV
-        echo "UPLOAD_ARTIFACT_NAME: $UPLOAD_ARTIFACT_NAME"
+    - name: Get reference commit for PR
+      if: github.event_name == 'pull_request'
+      env:
+        pull_request_title: ${{ github.event.pull_request.title }}
+      run: |
+        REFERENCE_SHA=${{ github.event.pull_request.base.sha }}
+        if [ -z "$REFERENCE_SHA" ]; then DO_RUN='false'; else DO_RUN='true'; fi
+        echo "REFERENCE_SHA=$REFERENCE_SHA" >> $GITHUB_ENV
+        echo "DO_RUN=$DO_RUN" >> $GITHUB_ENV
+        echo "PR_INFO= ${{ github.event.pull_request.number }} ${pull_request_title}" >> $GITHUB_ENV
+    - name: Get reference commit for push commit
+      if: github.event_name == 'push'
+      run: |
+        REFERENCE_SHA=$( git show -s --pretty=%P ${{ github.sha }} | head -c 10 )
+        if [ -z "$REFERENCE_SHA" ]; then DO_RUN='false'; else DO_RUN='true'; fi
+        echo "REFERENCE_SHA=$REFERENCE_SHA" >> $GITHUB_ENV
+        echo "DO_RUN=$DO_RUN" >> $GITHUB_ENV
+        echo "PR_INFO=" >> $GITHUB_ENV
+    - id: output_data
+      name: Output workflow data
+      env:
+        pr_info: ${{ env.PR_INFO }}
+      run: |
+        echo "::set-output name=do_run::${{ env.DO_RUN }}"
+        echo "DO_RUN=${{ env.DO_RUN }}"
+        echo "::set-output name=reference_sha::${{ env.REFERENCE_SHA }}"
+        echo "REFERENCE_SHA=${{ env.REFERENCE_SHA }}"
+        UPLOAD_ARTIFACT_NAME="$(tr '":<>|*?/\\' '_' <<<"VTests Comparison ${{ env.BUILD_NUMBER }}${pr_info}")"
+        echo "::set-output name=artifact_name::$UPLOAD_ARTIFACT_NAME"
+        echo "UPLOAD_ARTIFACT_NAME=$UPLOAD_ARTIFACT_NAME"
 
+  generate_current_pngs:
+    name: "Generate current PNGs"
+    runs-on: ubuntu-20.04
+    needs: setup
+    if: needs.setup.outputs.do_run == 'true' # Can't use env: see https://github.com/actions/runner/issues/480
+    steps:
+    - name: Clone repository
+      uses: actions/checkout@v2
```

Comment:
All these should be v3 too (just everything with `uses: actions/…`

Reply:
fixed